### ISSUE TITLE
Version Fetch Major.Minor

### ISSUE THREAD

**Issue by ax3l** [spack/spack#4373]
For short-hand version specifications such as

```bash
spack install openmpi@2
```

the *newest 2.x.y* release is not selected. Instead a warning of the following form is thrown:

```
==> Warning: There is no checksum on file to fetch openmpi@2 safely.
==> Error: FetchError: Will not fetch openmpi@2
    Add a checksum or use --no-checksum to skip this check.
```

For writing an install recipe, e.g. to partition a docker image, I would find this helpful. Expending automatically to the latest known concrete version allows expressing "install latest of the selected major and minor version if new patch level fixes are available" but I will not blindly install "the latest version" as with `spack install openmpi` since such might break the API compatibility.

Did I miss the syntax for that? Of course, there is always `openmpi@2+:2.999`, right? ;)

**Comment by adamjstewart** [comment]
I want this **so badly**. It works fine for specifying compiler versions, I install `cuda%gcc@4` all the time. But it doesn't work at all for packages. Being able to say that a package `depends_on('python@2.7')` or that I want to build `python@3` would be clutch. 

In the meantime, you can do something like `spack install python@3:` or `spack spec openmpi@:2.999`.

**Comment by becker33** [comment]
This was fixed in #36273
